web/src/OverviewItemView.tsx

```diff
@@ -484,15 +532,27 @@ export function OverviewItemDetails(props: OverviewItemDetailsProps) {
   }
 
   let pathBuilder = usePathBuilder()
-  let width = props.width || 330
   let isBuildingClass =
     item.buildStatus === ResourceStatus.Building ? "isBuilding" : ""
+  let ref = useRef(null as any)
+
+  // Simulate an 'expansion' animation by modifying the height post-render.
+  useEffect(() => {
+    if (ref.current && height) {
+      ref.current.style.height = ref.current.firstChild.scrollHeight + "px"
+    }
+  }, [height])
+
+  let widthStyle = width ? width + "px" : "330px"
```

Comment:
Minor question: Since this default width is referenced in multiple places within this React component, do you think we should have a var or const somewhere in here that declares the 330px? To make it slightly harder to change the value in one place but leave incorrect calculations elsewhere?

Previously, I stored these values in style-helper, but I do see the sense in confining vars only used within a single component in that same component.

Reply:
this is a good question!

The long-ish answer is that there are two different components here:
(1) OverviewItemView, which has a dynamic width based on the current screen size and the number of other cards
(2) OverviewItemDetail, which must be adjusted dynamically to match OverviewItemView

but since we want to view both of these components in storybooks (outside the context of the stuff that's determining their size), we use 330px in a bunch of places as a nice 'test' fixture size, but not an authoritative value that you'd want to keep in-sync.

But maybe the answer is to move test values into test files, so they don't accidentally get used as real values.

Reply:
Ahhh, I see. Yeah, in that case it would have helped me to understand that 330px was really just for the test fixture. Upon a quick read, I made the incorrect assumption that it was a "real" default for the actual component. So, I vote for a convention of keeping test values in test files.